extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,249 @@
+    );
+  }
+
+  /**
+   * Sets wado headers for all clients before queries
+   * @returns {void}
+   */
+  public setWadoHeaders(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.getWadoHeader(client))
+    );
+  }
+
+  /**
+   * Sets authorization headers for wado clients before queries
+   * @returns {void}
+   */
+  public setAuthorizationHeadersForWADO(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  /**
+   * Returns a boolean indicating if a client have reject abilities
+   * @returns {boolean} client reject support
+   */
+  public clientCanReject(name) {
+    return this.getClient(name)?.supportsReject;
+  }
+
+  /**
+   * Returns the reject function of a client
+   * @param name
+   * @returns {object} client reject object
+   */
+  public getClientRejectObject(name) {
+    const client = this.getClient(name);
+    if (client?.supportsReject) {
+      return dcm4cheeReject(client.wadoRoot);
+    }
+  }
+
+  /**
+   * Returns the qido client
+   * @param name
+   * @returns {object} qido client
+   */
+  public getQidoClient(name?: string): object {
+    const client = this.getClient(name);
+    return client?.qidoDicomWebClient;
+  }
+
+  /**
+   * Returns the wado client
+   * @param name
+   * @returns {object} wado client
+   */
+  public getWadoClient(name?: string): object {
+    const client = this.getClient(name);
+    return client?.wadoDicomWebClient;
+  }
+
+  /**
+   * Returns the client configuration
+   * @param name
+   * @returns {object} client configuration
+   */
+  public getConfig(name?: string): object {
+    return name ? this.clients.find(client => client.name === name) : this.clients[0];
+  }
+
+  /**
+   * Gets the client list already setting the necessary wado headers
+   * @returns {Array} client list
+   */
+  public getClientsForWadoRequests() {
+    this.setWadoHeaders();
+    return this.getClients();
+  }
+
+  /**
+   * Gets the client list already setting the necessary qido headers
+   * @returns {Array} client list
+   */
+  public getClientsForQidoRequests() {
+    this.setQidoHeaders();
+    return this.getClients();
+  }
+
+  /**
+   * Returns the client list
+   * @returns {Array} client list
+   */
+  public getClients() {
+    return this.clients;
+  }
+  /**
+   * Adds a client to client list given a dicomweb server configuration
+   * @param configToAdd
+   * @returns {void}
+   */
+  private addClient(configToAdd: ConfigToAdd): void {
+    // if no qidoRoot or wadoRoot, don't add the client. Could be the case for
+    // configurations that relies on onConfiguration function but lacks necessary
+    // additional information
+    if (!configToAdd?.qidoRoot || !configToAdd?.wadoRoot) {
+      return;
+    }
+    const config = Object.assign({}, configToAdd);
+    config.qidoConfig = {
```

Comment:
grab the headers above 

```js
const headers = this.userAuthenticationService.getAuthorizationHeader()
```

and use them, this is not related to each call to be safe or not, this is just client creation